config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml

```diff
@@ -98,3 +98,20 @@ presubmits:
         - config/jobs
     annotations:
       testgrid-create-test-group: 'true'
+  - name: pull-test-infra-validate-manifests
+    always_run: true
+    decorate: true
+    annotations:
+      testgrid-create-test-group: 'true'
+    spec:
+      # Kubectl needs access the discovery info from the kube api in order
+      # to know the schema used for validation
+      automountServiceAccountToken: true
+      containers:
+      - image: gcr.io/google-containers/hyperkube:v1.13.9
+        command:
+        - kubectl
+        - create
+        - --dry-run=true
```

Comment:
Is there an equivalent `bazel` invocation for the `production.apply` target?

Reply:
That is a question I can not answer

Reply:
@fejta @Katharine would love your input here!

Reply:
/assign fejta
assigning fejta so it shows up on his PR dashboard :)

Reply:
`bazel run //prow/cluster:production.apply -- --dry-run=true`

Reply:
@fejta Tried that, it fails:
```
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
Loading: 
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Analyzing: target //prow/cluster:production.create (1 packages loaded, 0 targets configured)
Analyzing: target //prow/cluster:production.create (12 packages loaded, 18 targets configured)
DEBUG: /root/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/io_bazel_rules_k8s/toolchains/kubectl/kubectl_toolchain.bzl:28:9: No kubectl tool was found or built, executing run for rules_k8s targets might not work.
Analyzing: target //prow/cluster:production.create (76 packages loaded, 6867 targets configured)
Analyzing: target //prow/cluster:production.create (76 packages loaded, 6867 targets configured)
Analyzing: target //prow/cluster:production.create (93 packages loaded, 6968 targets configured)
Analyzing: target //prow/cluster:production.create (93 packages loaded, 6968 targets configured)
DEBUG: /root/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/bazel_gazelle/internal/go_repository.bzl:184:13: gazelle: gazelle: finding module path for import github.com/moby/buildkit/api/services/control: exit status 1: go: finding github.com/moby/buildkit v0.6.2
go: downloading github.com/moby/buildkit v0.6.2
go: extracting github.com/moby/buildkit v0.6.2
go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/client: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/util/appcontext: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/util/progress/progressui: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.6.2 requires
	github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.3.0-0.20190507210959-7c1e88399ec0.mod: 410 Gone
gazelle: finding module path for import github.com/docker/swarmkit/api/defaults: exit status 1: can't load package: package github.com/docker/swarmkit/api/defaults: module github.com/docker/swarmkit@latest (v1.12.0) found, but does not contain package github.com/docker/swarmkit/api/defaults
gazelle: finding module path for import github.com/docker/swarmkit/api/genericresource: exit status 1: can't load package: package github.com/docker/swarmkit/api/genericresource: module github.com/docker/swarmkit@latest (v1.12.0) found, but does not contain package github.com/docker/swarmkit/api/genericresource
DEBUG: /root/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/bazel_gazelle/internal/go_repository.bzl:184:13: gazelle: gazelle: /root/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/com_github_docker_docker_credential_helpers/secretservice/secretservice_linux.go: error reading go file: /root/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/com_github_docker_docker_credential_helpers/secretservice/secretservice_linux.go: pkg-config not supported: #cgo pkg-config: libsecret-1
INFO: Analyzed target //prow/cluster:production.create (99 packages loaded, 7073 targets configured).
INFO: Found 1 target...
[0 / 73] [Prepa] BazelWorkspaceStatusAction stable-status.txt ... (7 actions, 0 running)
Target //prow/cluster:production.create up-to-date:
  bazel-bin/prow/cluster/production.create
INFO: Elapsed time: 171.534s, Critical Path: 6.40s
INFO: 94 processes: 94 processwrapper-sandbox.
INFO: Build completed successfully, 138 total actions
INFO: Running command line: bazel-bin/prow/cluster/production.create '--dry-run=true'
INFO: Build completed successfully, 138 total actions
kubectl toolchain was not properly configured so boskos.create cannot be executed.
kubectl toolchain was not properly configured so boskos-janitor.create cannot be executed.
kubectl toolchain was not properly configured so boskos-metrics.create cannot be executed.
kubectl toolchain was not properly configured so boskos-reaper_deployment.create cannot be executed.
kubectl toolchain was not properly configured so build_deployment.create cannot be executed.
kubectl toolchain was not properly configured so build_rbac.create cannot be executed.
kubectl toolchain was not properly configured so cert-manager.create cannot be executed.
kubectl toolchain was not properly configured so cluster-issuer.create cannot be executed.
kubectl toolchain was not properly configured so crier_deployment.create cannot be executed.
kubectl toolchain was not properly configured so crier_rbac.create cannot be executed.
kubectl toolchain was not properly configured so deck_service.create cannot be executed.
kubectl toolchain was not properly configured so deck_deployment.create cannot be executed.
kubectl toolchain was not properly configured so gce-ssd-retain_storageclass.create cannot be executed.
kubectl toolchain was not properly configured so ghproxy.create cannot be executed.
kubectl toolchain was not properly configured so grandmatriarch.create cannot be executed.
kubectl toolchain was not properly configured so hook_service.create cannot be executed.
kubectl toolchain was not properly configured so hook_deployment.create cannot be executed.
kubectl toolchain was not properly configured so horologium_deployment.create cannot be executed.
kubectl toolchain was not properly configured so mem-limit-range_limitrange.create cannot be executed.
kubectl toolchain was not properly configured so needs-rebase_deployment.create cannot be executed.
kubectl toolchain was not properly configured so pipeline_deployment.create cannot be executed.
kubectl toolchain was not properly configured so plank_service.create cannot be executed.
kubectl toolchain was not properly configured so plank_deployment.create cannot be executed.
kubectl toolchain was not properly configured so prowjob_customresourcedefinition.create cannot be executed.
kubectl toolchain was not properly configured so pushgateway_deployment.create cannot be executed.
kubectl toolchain was not properly configured so sinker_service.create cannot be executed.
kubectl toolchain was not properly configured so sinker_deployment.create cannot be executed.
kubectl toolchain was not properly configured so statusreconciler_deployment.create cannot be executed.
kubectl toolchain was not properly configured so tide_service.create cannot be executed.
kubectl toolchain was not properly configured so tide_deployment.create cannot be executed.
kubectl toolchain was not properly configured so tls-ing_ingress.create cannot be executed.
kubectl toolchain was not properly configured so tot_service.create cannot be executed.
kubectl toolchain was not properly configured so tot_deployment.create cannot be executed.
kubectl toolchain was not properly configured so tune-sysctls_daemonset.create cannot be executed.
```

Reply:
Try with https://github.com/kubernetes/test-infra/pull/14863? This provides a hermetic kubectl

Reply:
Okay, I do not have time to extensively test this via mkpj right now, WDYT about marking it optional, merging this and then checking on another PR if it works, if yes making it mandatory?

Reply:
Given that this is a test can't we just make a test Target for this? Then it will automatically run for people running bazel test (including CI)

I don't think this warrants its own context.

Reply:
Sure, if you can give me something to read up on how to do that.

What should be mentioned thought is that a kube apiserver needs to be available, because `kubectl` fetches the openapi schema used for validation from it